docs/config/lua/wezterm.table/merge.md

```diff
@@ -0,0 +1,36 @@
+# `wezterm.table.merge(array_of_tables [, keep_first])`
+
+{{since('nightly')}}
+
+This function merges Lua tables passed to it in the form of an array.
```

Comment:
It wasn't immediately obvious to me how the merge might take place, so let's try to make it clear up-front:

```suggestion
This function merges a list of Lua object-style tables based on their keys.

The tables are passed to it in the form of an array.
```